cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/RemoteSourceControlOperationRunner.java

```
@@ -0,0 +1,98 @@
+    HttpRequestConfig httpRequestConfig = new HttpRequestConfig(connectTimeout, readTimeout, false);
+    this.remoteTaskExecutor = new RemoteTaskExecutor(cConf, metricsCollectionService, remoteClientFactory,
+                                                     RemoteTaskExecutor.Type.TASK_WORKER, httpRequestConfig);
+  }
+
+  @Override
+  public PushAppResponse push(PushAppContext pushAppContext)
+    throws PushFailureException, NoChangesToPushException, AuthenticationConfigException {
+    try {
+      RunnableTaskRequest request =
+        RunnableTaskRequest.getBuilder(PushAppTask.class.getName()).withParam(GSON.toJson(pushAppContext)).build();
+      
+      LOG.debug("Pushing application {} to linked repository", pushAppContext.getAppToPush());
+      byte[] result = remoteTaskExecutor.runTask(request);
+      return GSON.fromJson(new String(result, StandardCharsets.UTF_8), PushAppResponse.class);
+    } catch (RemoteExecutionException e) {
+      RemoteTaskException remoteTaskException = e.getCause();
+      String exceptionClass = remoteTaskException.getRemoteExceptionClassName();
+      String exceptionMessage = remoteTaskException.getMessage();
+
+      if (NoChangesToPushException.class.getName().equals(exceptionClass)) {
+        throw new NoChangesToPushException(exceptionMessage);
+      }
+
+      if (AuthenticationConfigException.class.getName().equals(exceptionClass)) {
+        throw new AuthenticationConfigException(exceptionMessage, remoteTaskException.getCause());
+      }
+
+      throw new PushFailureException(exceptionMessage, remoteTaskException.getCause());
+    } catch (Exception ex) {
+      throw new PushFailureException(ex.getMessage(), ex.getCause());
+    }
+  }
+
+  @Override
+  public PullAppResponse<?> pull(ApplicationReference appRef, RepositoryConfig repoConfig)
+    throws PullFailureException, NotFoundException, AuthenticationConfigException {
+    // TODO: CDAP-20356, pull application in task worker
+    return null;
```

It's better to throw an `UnsupportedOperationException` as it allows to fail early

---

Throw `UnsupportedOperationException` now